core/src/main/java/com/digitalpebble/storm/crawler/bolt/ParserBolt.java

```diff
@@ -238,8 +242,46 @@ public void execute(Tuple tuple) {
         LOG.info("Parsed {} in {} msec", url, duration);
 
         // apply the parse filters if any
-        parseFilters.filter(url, content, root, metadata);
+        try {
+            parseFilters.filter(url, content, root, metadata);
+        } catch (RuntimeException e) {
+            String errorMessage = "Exception while running parse filters on "
```

Comment:
Because my comment got lost, I'll add it back here: The Java compiler uses a StringBuilder when concatenating multiple Strings. So for readability purposes, I prefer using the `+` sign. As far as the getMessage() being null, this has been addressed by using toString instead (implied by the String concatenation)